test/kafka-exactly-once/before-restart.td

```diff
@@ -68,6 +68,15 @@ $ kafka-create-topic topic=input-consistency
 
```

Comment:
I think your changes to the test provide what mine did (see below). Should I push them to your branch or they are basically testing the same thing?

```
diff --git a/test/kafka-exactly-once/before-restart.td b/test/kafka-exactly-once/before-restart.td
index 27d8ccb66..9901567d4 100644
--- a/test/kafka-exactly-once/before-restart.td
+++ b/test/kafka-exactly-once/before-restart.td
@@ -7,6 +7,8 @@
 # the Business Source License, use of this software will be governed
 # by the Apache License, Version 2.0.
 
+$ set-regex match=\d{13} replacement=<TIMESTAMP>
+
 $ set schema={
     "type": "record",
     "name": "envelope",
@@ -73,9 +75,18 @@ $ kafka-create-topic topic=input
     WITH (consistency_topic = 'testdrive-input-consistency-${testdrive.seed}')
   FORMAT AVRO USING SCHEMA '${schema}' ENVELOPE DEBEZIUM
 
-> CREATE SINK output FROM input_byo
-  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'output-sink-${testdrive.seed}'
-  WITH (exactly_once=true, consistency_topic='output-sink-consistency-${testdrive.seed}')
+> CREATE MATERIALIZED SOURCE input_ts
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-input-${testdrive.seed}'
+  FORMAT AVRO USING SCHEMA '${schema}' ENVELOPE DEBEZIUM
+
+> CREATE SINK output_byo FROM input_byo
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'output-byo-sink-${testdrive.seed}'
+  WITH (exactly_once=true, consistency_topic='output-byo-sink-consistency-${testdrive.seed}')
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+
+> CREATE SINK output_ts FROM input_ts
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'output-ts-sink-${testdrive.seed}'
+  WITH (exactly_once=true, consistency_topic='output-ts-sink-consistency-${testdrive.seed}')
   FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
 
 $ kafka-ingest format=avro topic=input-consistency timestamp=1 schema=${trxschemakey}
@@ -105,7 +116,7 @@ $ kafka-ingest format=avro topic=input schema=${schema} timestamp=4
 {"before": null, "after": {"row": {"a": 3, "b": 4}}}
 {"before": null, "after": {"row": {"a": 5, "b": 6}}}
 
-$ kafka-verify format=avro sink=materialize.public.output
+$ kafka-verify format=avro sink=materialize.public.output_byo
 {"before": null, "after": {"row": {"a": 1, "b": 1}}, "transaction": {"id": "1"}}
 {"before": null, "after": {"row": {"a": 1, "b": 2}}, "transaction": {"id": "1"}}
 {"before": null, "after": {"row": {"a": 2, "b": 1}}, "transaction": {"id": "1"}}
@@ -115,6 +126,16 @@ $ kafka-verify format=avro sink=materialize.public.output
 {"before": null, "after": {"row": {"a": 3, "b": 4}}, "transaction": {"id": "3"}}
 {"before": null, "after": {"row": {"a": 5, "b": 6}}, "transaction": {"id": "3"}}
 
+$ kafka-verify format=avro sink=materialize.public.output_ts
+{"before":null,"after":{"row":{"a":1,"b":1}},"transaction":{"id":"<TIMESTAMP>"}}
+{"before":null,"after":{"row":{"a":1,"b":2}},"transaction":{"id":"<TIMESTAMP>"}}
+{"before":null,"after":{"row":{"a":2,"b":1}},"transaction":{"id":"<TIMESTAMP>"}}
+{"before":null,"after":{"row":{"a":3,"b":1}},"transaction":{"id":"<TIMESTAMP>"}}
+{"before":null,"after":{"row":{"a":3,"b":4}},"transaction":{"id":"<TIMESTAMP>"}}
+{"before":null,"after":{"row":{"a":5,"b":6}},"transaction":{"id":"<TIMESTAMP>"}}
+{"before":null,"after":{"row":{"a":11,"b":11}},"transaction":{"id":"<TIMESTAMP>"}}
+{"before":null,"after":{"row":{"a":22,"b":11}},"transaction":{"id":"<TIMESTAMP>"}}
+
 > SELECT * FROM input_byo
 a  b
 ------
@@ -126,3 +147,15 @@ a  b
 22  11
 3   4
 5   6
+
+> SELECT * FROM input_ts
+a  b
+------
+1   1
+2   1
+3   1
+1   2
+11  11
+22  11
+3   4
+5   6
```

Reply:
oh man I wish I had known about `set-regex` earlier